### DISCUSSION TITLE
How to specify a custom security scheme name

### DISCUSSION THREAD

**Discussion by forik** [microsoft/typespec#4774] [Q&A] [answered]
I have an example typespec:
```
import "@typespec/http";
using TypeSpec.Http;

alias MyOAuth2<Scopes extends string[]> = OAuth2Auth<
  [
    {
      type: OAuth2FlowType.clientCredentials;
      tokenUrl: "https://example.com/auth/token";
    }
  ],
  Scopes
>;

@useAuth(MyOAuth2<["123"]>)
op get(): void;

@useAuth(MyOAuth2<["1234"]>)
@route("/1")
op get2(): Resp;

@doc("Test")
model Resp {
  @statusCode code: 200;
}
```
This produces the following:
```
openapi: 3.0.0
info:
  title: (title)
  version: 0.0.0
tags: []
paths:
  /:
    get:
      operationId: get
      parameters: []
      responses:
        '204':
          description: 'There is no content to send for this request, but the headers may be useful. '
      security:
        - OAuth2Auth:
            - '123'
  /1:
    get:
      operationId: get2
      parameters: []
      responses:
        '200':
          description: Test
      security:
        - OAuth2Auth:
            - '1234'
components:
  securitySchemes:
    OAuth2Auth:
      type: oauth2
      flows:
        clientCredentials:
          tokenUrl: https://sandbox.upvest.co/auth/token
          scopes:
            '123': ''
            '1234': ''
```

How to change `OAuth2Auth` to a custom value?
And related follow up question: How to add a scope description to `components/securitySchemes/scheme/flows/scopes`?

**Comment by timotheeguerin** [comment, marked as answer]
If you use `model is` instead you can achieve that
[playground](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CnVzaW5nIFR5cGVTcGVjLkh0dHA7Cgptb2RlbCBNeU9BdXRoMjxTY29wZXMgZXh0ZW5kcyBzdHJpbmdbXT4KICBpcyDGJcQFPAogICAgW8UGICB7xwggIMR1OscoRmxvd8RvLmNsaWVudENyZWRlbnRpYWxzO8owb2tlblVybDogIuQAp3M6Ly9leGFtcGxlLmNvbS9hdXRoL8UkIsg0fcUIXSzFB%2BYArwogID47CgpAdXNlxHko6QDOWyIxMjMiXT4pCm9wIGdldCgpOiB2b2lk2i00xS5Acm91dGUoIi8xIsg7MsQ8UmVz5AE0QGRvYygiVGVzdCIp5wFBxBrlAQdAc3RhdHVzQ29kZSBjb2RlOiAyMDA7Cn0K&e=%40typespec%2Fopenapi3&options=%7B%7D) 
```ts

model MyOAuth2<Scopes extends string[]>
  is OAuth2Auth<
    [
      {
        type: OAuth2FlowType.clientCredentials;
        tokenUrl: "https://example.com/auth/token";
      }
    ],
    Scopes
  >;
```

**Reply by forik** [reply]
So the scheme name would be taken from model name. That works as well probably. Thanks

**Reply by a-akshat** [reply]
@timotheeguerin thank you for your response.
I wanted to see if there is a way we can add description to the accepted scopes in a flow. OPs second question.
Let me know if you want me to start a separate issue for the same.
